src/migrations/1666714308809-single-card-callouts.ts

```diff
@@ -0,0 +1,172 @@
+        queryRunner,
+        callout.id
+      );
+
+      if (cardType) {
+        // cards are from the same type
+        await createSingleCardTypeTemplate(queryRunner, callout.id, cardType);
+      } else {
+        // cards are from different types
+        await createContributionTemplate(queryRunner, callout.id);
+      }
+    }
+    // callouts without any cards & nameID != 'card-default'
+    const calloutsWithoutCards: { id: string }[] = await queryRunner.query(`
+          SELECT id FROM callout
+          WHERE id NOT IN (
+            SELECT DISTINCT callout.id AS calloutId FROM aspect
+            LEFT JOIN callout ON aspect.calloutId = callout.id
+          )
+          AND nameID NOT LIKE 'card-default%'
+        `);
+    // create a card template
+    for (const callout of calloutsWithoutCards) {
+      await createOtherTemplate(queryRunner, callout.id);
+    }
+    // select callouts with multiple cards & nameID != 'card-default'
+    const multipleCardCallout: { id: string }[] = await queryRunner.query(`
+          SELECT id FROM callout
+          WHERE id IN (
+            SELECT callout.id AS calloutId FROM aspect
+            LEFT JOIN callout ON aspect.calloutId = callout.id
+          )
+          AND nameID NOT LIKE 'card-default%'
+        `);
+    // run query per callout if all it's cards are the same type
+    for (const callout of multipleCardCallout) {
+      const cardType = await getSingleCardTypeFromCallout(
+        queryRunner,
+        callout.id
+      );
+
+      if (cardType) {
+        await createSingleCardTypeTemplate(queryRunner, callout.id, cardType);
+      } else {
+        await createOtherTemplate(queryRunner, callout.id);
+      }
+    }
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE \`callout\`  DROP FOREIGN KEY \`FK_22a2ec1b5bca6c54678ffb19eb0\``
+    );
+    await queryRunner.query(
+      `DROP INDEX \`REL_22a2ec1b5bca6c54678ffb19eb\` ON \`callout\``
+    );
+    await queryRunner.query(
+      `ALTER TABLE \`callout\` DROP INDEX \`IDX_22a2ec1b5bca6c54678ffb19eb\``
+    );
+    await queryRunner.query(
+      `ALTER TABLE \`callout\` DROP COLUMN \`cardTemplateId\``
+    );
+  }
+}
+
+const getOtherTemplateVisualUri = async (
+  runner: QueryRunner,
+  calloutId: string
+): Promise<string> =>
+  runner
+    .query(
+      `
+        SELECT visual.uri FROM callout
+        LEFT JOIN collaboration ON collaboration.id = callout.collaborationId
+        LEFT JOIN hub ON hub.collaborationId = collaboration.id
+        LEFT JOIN templates_set ON templates_set.id = hub.templatesSetId
+        LEFT JOIN aspect_template ON aspect_template.templatesSetId = templates_set.id
+        LEFT JOIN template_info ON template_info.id = aspect_template.templateInfoId
+        LEFT JOIN visual ON visual.id = template_info.visualId
+        WHERE template_info.title = 'other' AND callout.id = '${calloutId}'
```

Comment:
Title can be changed from the client. Template type on the other hand cannot. Use `aspect_template.type` instead of `template_info.title`